funread_backend/Badges/views.py

```diff
@@ -1,3 +1,180 @@
+from django.forms import CharField
 from django.shortcuts import render
+from django.shortcuts import get_object_or_404
+from rest_framework.response import Response
+from rest_framework.decorators import api_view
+from rest_framework import status
+from .models import Badge
+from .serializers import BadgeSerializer
+import verifyJwt
+from .models import Badge
+from UserBadge.models import UserBadge
+from django.db.models import OuterRef, Subquery, BooleanField, Case, When, Value, IntegerField, CharField
+from django.http import JsonResponse
+from User_Levels.models import UserLevels
 
-# Create your views here.
+
+#Crear una nueva insignia
+@api_view(['POST'])
+def create_badge(request):
+    try:
+        authorization_header = request.headers.get('Authorization')
+        verify = verifyJwt.JWTValidator(authorization_header)
+        es_valido = verify.validar_token()
+        if es_valido==False:
+         return Response(status=status.HTTP_401_UNAUTHORIZED)
+        
+        # Validar los datos de la solicitud usando el serializador
+        serializer = BadgeSerializer(data=request.data)
+        
+        # Verificar si los datos son válidos
+        if serializer.is_valid():
+            serializer.save()  # Guardar la nueva insignia en la base de datos
+            return Response(serializer.data, status=status.HTTP_201_CREATED)
+        
+        # Si los datos no son válidos, devolver los errores
+        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
```

Comment:
Usemos negaciones para evitar nested statements
```suggestion
        if serializer.is_valid() == False:
           return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
             
        serializer.save()  # Guardar la nueva insignia en la base de datos
        return Response(serializer.data, status=status.HTTP_201_CREATED)
```